pkg/paravirt/paravirt_protected_entity.go

```diff
@@ -117,22 +118,17 @@ func (this ParaVirtProtectedEntity) DeleteSnapshot(
 	snapshotToDelete astrolabe.ProtectedEntitySnapshotID,
 	params map[string]map[string]interface{}) (bool, error) {
 	this.logger.Infof("ParaVirtProtectedEntity: DeleteSnapshot called on Paravirtualized Protected Entity, %v snapshotToDelete: %s", this.id.String(), snapshotToDelete.String())
-	peInfo, err := this.GetInfo(ctx)
-	if err != nil {
-		this.logger.Errorf("Failed to get info for ParaVirtProtectedEntity %v", this.id.String())
-		return false, errors.WithStack(err)
-	}
-	this.logger.Infof("ParaVirtProtectedEntity: Retrieved info id: %s, name: %s", peInfo.GetID().String(), peInfo.GetName())
-
+	var err error
 	backupRepositoryName, ok := params[astrolabe.PvcPEType]["BackupRepositoryName"].(string)
 	if !ok {
 		backupRepositoryName = "INVALID_BR_NAME"
 	}
+	pvPeIdComponents := strings.Split(this.id.String(), ":")
```

Comment:
Let's not treat the ID as a string and assume the internals.  We have accessors.  If you don't have an accessor for what you want here, add it to the ID object.  If we change the string representation this code may break.